server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java

```diff
@@ -134,7 +135,17 @@ public void doPost(final HttpServletRequest request, final HttpServletResponse r
 
             final FileItemFactory fileItemFactory = new DiskFileItemFactory();
             final ServletFileUpload serviceFileUpload = createServletFileUpload(fileItemFactory);
-            final List<FileItem> items = serviceFileUpload.parseRequest(request);
+            if (checkUploadedFileSize) {
+                setUploadSizeMax(serviceFileUpload, request);
+            }
+            List<FileItem> items;
+            try {
+                items = serviceFileUpload.parseRequest(request);
+            } catch (final OutOfMemoryError e) {
+                throw new SizeLimitExceededException("The file exceeds its maximum permitted size.",
```

Comment:
I feel that the erro message is not correct. it is not that it exceeds its permitted size but the file exceeds the current server capacity.
The permitted size shoudl be handled by setUploadSizeMax( (if I understood well)